haystack/components/joiners/document_joiner.py

```diff
@@ -149,7 +162,8 @@ def run(self, documents: Variadic[List[Document]], top_k: Optional[int] = None):
 
         return {"documents": output_documents}
 
-    def _concatenate(self, document_lists: List[List[Document]]) -> List[Document]:
+    @staticmethod
```

Comment:
If we change this method to static, when should change `self._concatenate` to `DocumentJoiner._concatenate` when we call it.  https://github.com/deepset-ai/haystack/pull/8593/files#diff-2e14ed3730889db5864049422ba67743ccb9fd06a20e271a387b423c2eb8ab12R260
Also, distribution_based_rank_fusion should be static too then.
I suggest to keep `_concatenate` as is for this PR so that we can quickly merge it.

Reply:
good point, I will open a separate PR for those other changes